aesara/tensor/shape.py

```diff
@@ -537,29 +534,31 @@ def specify_shape(
         int, List[Union[int, Variable]], Tuple[Union[int, Variable]], Variable
     ],
 ):
-    """Specify a fixed shape for a `Variable`."""
+    """Specify a fixed shape for a `Variable`.
```

Comment:
Do we need this helper function? Can't the code go into make_node?

Reply:
Not if it would make the `Op.make_node` and `Apply.inputs` inconsistent.  It looks like this `specify_shape` helper/constructor function has an `(x, shape)` signature and not `(x, *shape)`, so that might be one reason not to get rid of it.

Reply:
My main concern was to make sure that `SpecifyShape` `Op` would work with it's own inputs so that you can call `node.owner.op(*node.owner.inputs)`. I guess once we have that, it doesn't matter we have this helper logic separated.

Reply:
> My main concern was to make sure that `SpecifyShape` `Op` would work with it's own inputs so that you can call `node.owner.op(*node.owner.inputs)`.

Yes, that is by far the most important thing.

More generally, it would be great if we could confirm this automatically somehow (e.g. via type checking, but that doesn't seem likely).